File: Bits/puissanceDeDeux.js
/**
 * Vérifier si un nombre est une puissance de deux
 * @param {number} int - Nombre à vérifier
 * @returns {boolean} - True si le nombre est un puissance de deux, false sinon
 */
const estPuissanceDeDeux = int => (int & (int - 1)) === 0;

// Test
console.log(estPuissanceDeDeux(8)); // true
console.log(estPuissanceDeDeux(7)); // false

/*
On prend un nombre entier n, on vérifie si n & (n - 1) est égal à 0.
Si c'est le cas, alors n est un nombre de puissance de deux.
00000000000000000000000000001000 = 8
00000000000000000000000000000111 = 7
8 & (8 - 1) = 0, donc 8 est un nombre de puissance de deux.
*/

File: Go/Bits/puissanceDeDeux.go
package main

import "fmt"

func puissanceDeDeux(n int) bool {
	return n&(n-1) == 0
}

func main() {
	cases := []int{1, 2, 3, 4, 16, 31, 64}
	for _, c := range cases {
		fmt.Printf("%d est une puissance de deux = %t\n", c, puissanceDeDeux(c))
	}
}

//Output:
// 1  est une puissance de deux = true
// 2  est une puissance de deux = true
// 3  est une puissance de deux = false
// 4  est une puissance de deux = true
// 16 est une puissance de deux = true
// 31 est une puissance de deux = false
// 64 est une puissance de deux = true
